api/src/transactions.rs

```diff
@@ -530,10 +578,10 @@ impl TransactionsApi {
         ))
     }
 
-    // This function looks for the transaction by hash in database and then mempool,
-    // because the period a transaction stay in the mempool is likely short.
-    // Although the mempool get transation is async, but looking up txn in database is a sync call,
-    // thus we keep it simple and call them in sequence.
+    /// Retrieves a transaction by hash in database and then mempool,
+    /// because the period a transaction stay in the mempool is likely short.
+    /// Although the mempool get transation is async, but looking up txn in database is a sync call,
+    /// thus we keep it simple and call them in sequence.
```

Comment:
This could do with a rewrite, how about:

Retrieves a transaction by hash. First the node tries to find the transaction in the DB. If the transaction is found there, it means the transaction is committed. If it is not found there, it looks in mempool. If it is found there, it means the transaction is still pending.

Though we could still do better than that.